### PR TITLE
Sjh move tasks to own packages

### PR DESCRIPTION
**Pull request by stephenhand**
## Description

Refactor to split schedule task code out of the HRM service package.

* Move current code out of `hrm-service` into a new `hrm-core` package, that `hrm-service` depends on
* Move task code into new packages under `scheduled-tasks` that depend on `hrm-core`
* `hrm-service` now just pulls these modules together and exposes the bootstrap code, npm tasks and contains the service tests. Essentially this is responsible for building the monolithic artefact
* Migrated to automatically produce ISO strings from TIMESTAMP WITH TIMEZONE columns
* Small refactor of khp data pull to allow better type safety

The idea would be to split the HTTP element out of hrm-core in the future, so it's the pure business / data access logic, and the APIs are implemented elsewhere

I did NOT move the contact-job-processor out of hrm-core, just because we have plans to replace that in the future anyway

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added

### Related Issues
Fixes #....

### CODE REVIEW

**Review by murilovmachado — APPROVED** [review]
LGTM 🚀 

Maybe on the Monday's Engineering meeting you can talk about the changes you've done here and the intentions behind it.

File: hrm-domain/hrm-service/tsconfig.json [hunk 1]
```diff
@@ -4,10 +4,11 @@
     "module": "commonjs",
     "esModuleInterop": true,
     "target": "ES2020",
-    "moduleResolution": "node",
+    "moduleResolution": "node16",
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/tsconfig.json:7]: :thinking:

**Review by GPaoloni — APPROVED** [review]
This is an amazing refactor, thanks!!

> The idea would be to split the HTTP element out of hrm-core in the future, so it's the pure business / data access logic, and the APIs are implemented elsewhere

:raised_hands: :rocket: 
I'm assuming that code will be moved back into the `hrm-service`? If so, why not doing so in this iteration, was it too messy?

Comments/questions:
- Maybe `hrm-domain/hrm-core/app.ts` could/should live under the service instead?
- Can I ask more context on this? :thinking: 
  > I did NOT move the contact-job-processor out of hrm-core, just because we have plans to replace that in the future anyway

**Comment by stephenhand** [comment]
> This is an amazing refactor, thanks!!
> 
> > The idea would be to split the HTTP element out of hrm-core in the future, so it's the pure business / data access logic, and the APIs are implemented elsewhere
> 
> 🙌 🚀 I'm assuming that code will be moved back into the `hrm-service`? If so, why not doing so in this iteration, was it too messy?
> 
> Comments/questions:
> 
> * Maybe `hrm-domain/hrm-core/app.ts` could/should live under the service instead?
> * Can I ask more context on this? 🤔
>   > I did NOT move the contact-job-processor out of hrm-core, just because we have plans to replace that in the future anyway

I was thinking we would move the HTTP element into another separate module `hrm-http` - then we would have the same structure for everything using the HRM logic - each 'interface' (web api or task) is in it's own module that depends on hrm-core. Then hrm-service just pulls all these interfaces together into the single monolith that we currently deploy. hrm-service doesn't have any logic of its own beyond bootstrapping the constituent interfaces and providing some configuration

But yes, maybe move app.ts? Anything related to Express should eventually live in hrm-http though, ideally hrm-service would not direclty depend on Express, except possibly for types

**Comment by stephenhand** [comment]
> LGTM 🚀
> 
> Maybe on the Monday's Engineering meeting you can talk about the changes you've done here and the intentions behind it.

I will def do that

**Pull request merged**